src/core/blockchain/zilliqa/name-service.ts

```diff
@@ -21,15 +22,31 @@ export class NameService implements IBlockchainNameService {
                 name: ''
             });
         } else {
-            return Promise.reject({
-                error: ResolveTextError.INVALID
-            });
+            const { address } = await this.resolveName(text);
+            if (address === '0x0000000000000000000000000000000000000000' || address === '') {
+                return Promise.reject({
+                    error: ResolveTextError.INVALID
+                });
+            } else {
+                return Promise.resolve({
+                    code: ResolveTextCode.OK,
+                    type: ResolveTextType.NAME,
+                    address,
+                    name: ''
+                });
+            }
         }
     }
 
-    public resolveName(text: string): Promise<IResolveNameResponse> {
-        return Promise.resolve({
-            address: text
-        });
+    public async resolveName(name: string): Promise<IResolveNameResponse> {
+        const node = namehash(name);
+        const znsToBech32 = fromBech32Address(ZNS);
```

Comment:
the const name is toBech but you use fromBech :)

Reply:
Ah right, will change it